src/pip/_internal/index.py

```diff
@@ -877,7 +877,7 @@ def __init__(self, content, url, headers=None):
         self.headers = headers
 
     def __str__(self):
-        return self.url
+        return redact_password_from_url(self.url)
```

Comment:
@uranusjr Can you confirm whether it's okay for this to return a redacted URL?

Reply:
Actually I don’t think this method is called anywhere at all.

Reply:
Thanks! I guess it won't hurt to change it then.